doc: add how-to for bundling text resources (#8748)
ocaml/dune: doc/howto/bundle.rst

How to Bundle Resources
=======================

This guide will show you how to configure Dune to generate modules with string resources
from other files in your project.

Folder Structure
----------------

.. code:: bash

>     ❯ tree src

Reviewer: ```suggestion
    $ tree src
```

    src
    └── lib
        └── my_lib
            ├── dune
            └── resources

>                 └── site.css

Reviewer 2: IMO this can be simplified to
```
    lib
    ├── dune
    └── resources
        └── site.css
```
no need to invoke `tree`, or show two unnecessary levels.

Author: That's there for people, who like me, need an additional pointer on where the `resources` go in relation to the project root and the `dune` file. If you disagree with it though, feel free to create a PR removing it!

Dune Configuration
------------------

> See the section on :doc:`../reference/actions` for more details on ``progn`` and ``with-stdout-to``.

Reviewer: ```suggestion
See the section on :doc:`../reference/actions` for more details on :dune:ref:`action-progn` and :dune:ref:`action-with-<outputs>-to`.
```
